test/SILOptimizer/constant_propagation_stdlib.swift

```diff
@@ -0,0 +1,99 @@
+// CHECK-O:             return [[RESULT]]
+// CHECK-O:           } // end sil function '$s27constant_propagation_stdlib16isConcrete_falseyBi1_xlF'
+public func isConcrete_false<T>(_ x: T) -> Builtin.Int1 {
+  return Builtin.isConcrete(T.self)
+}
+
+// CHECK-ONONE-LABEL: sil @$s27constant_propagation_stdlib25isConcrete_generic_calleryBi1_xlF : $@convention(thin) <T> (@in_guaranteed T) -> Builtin.Int1 {
+// CHECK-ONONE:       bb0(
+// CHECK-ONONE:         debug_value_addr %0 : $*T, let, name "x", argno 1
+// CHECK-ONONE:         %2 = function_ref @$s27constant_propagation_stdlib16isConcrete_falseyBi1_xlF : $@convention(thin) <τ_0_0> (@in_guaranteed τ_0_0) -> Builtin.Int1
+// CHECK-ONONE:         [[RESULT:%.*]] = apply %2<T>(%0) : $@convention(thin) <τ_0_0> (@in_guaranteed τ_0_0) -> Builtin.Int1
+// CHECK-ONONE:         return [[RESULT]]
+// CHECK-ONONE:       } // end sil function '$s27constant_propagation_stdlib25isConcrete_generic_calleryBi1_xlF'
+// CHECK-O-LABEL:     sil @$s27constant_propagation_stdlib25isConcrete_generic_calleryBi1_xlF : $@convention(thin) <T> (@in_guaranteed T) -> Builtin.Int1 {
+// CHECK-O:           bb0(
+// CHECK-O:             [[GEN_FUNC:%.*]] = function_ref @$s27constant_propagation_stdlib16isConcrete_falseyBi1_xlFTf4d_n : $@convention(thin) <τ_0_0> () -> Builtin.Int1
+// CHECK-O:             [[RESULT:%.*]] = apply [[GEN_FUNC]]<T>() : $@convention(thin) <τ_0_0> () -> Builtin.Int1
+// CHECK-O:             return [[RESULT]]
+// CHECK-O:           } // end sil function '$s27constant_propagation_stdlib25isConcrete_generic_calleryBi1_xlF'
+public func isConcrete_generic_caller<T>(_ x: T) -> Builtin.Int1 {
+  return isConcrete_false(x)  
+}
+
+// CHECK-ONONE-LABEL: sil @$s27constant_propagation_stdlib26isConcrete_concrete_calleryBi1_AA5MyIntVF : $@convention(thin) (MyInt) -> Builtin.Int1 {
+// CHECK-ONONE:       bb0(
+// CHECK-ONONE:         debug_value %0 : $MyInt, let, name "x", argno 1
+// CHECK-ONONE:         %2 = alloc_stack $MyInt
+// CHECK-ONONE:         store %0 to %2 : $*MyInt
+// CHECK-ONONE:         %4 = function_ref @$s27constant_propagation_stdlib25isConcrete_generic_calleryBi1_xlF : $@convention(thin) <τ_0_0> (@in_guaranteed τ_0_0) -> Builtin.Int1
+// CHECK-ONONE:         [[RESULT:%.*]] = apply %4<MyInt>(%2) : $@convention(thin) <τ_0_0> (@in_guaranteed τ_0_0) -> Builtin.Int1
+// CHECK-ONONE:         dealloc_stack %2 : $*MyInt
+// CHECK-ONONE:         return [[RESULT]]
+// CHECK-ONONE:       } // end sil function '$s27constant_propagation_stdlib26isConcrete_concrete_calleryBi1_AA5MyIntVF'
+// CHECK-O-LABEL:     sil @$s27constant_propagation_stdlib26isConcrete_concrete_calleryBi1_AA5MyIntVF : $@convention(thin) (MyInt) -> Builtin.Int1 {
+// CHECK-O:           bb0(
+// CHECK-O:             [[RESULT:%.*]] = integer_literal $Builtin.Int1, -1
+// CHECK-O:             return [[RESULT]]
+// CHECK-O:           } // end sil function '$s27constant_propagation_stdlib26isConcrete_concrete_calleryBi1_AA5MyIntVF'
```

Comment:
I'm a little bit concerned that the patterns we're matching here include a lot of stuff that isn't directly part of isConcrete; my gut feeling is that these tests are probably going to be pretty fragile. Do any of the people who work on SIL have opinions about this?

Reply:
I have a pretty strong opinion that checks against generated SIL should be minimal. They need to be just specific enough not to allow false negatives and no more specific. Rewriting SIL tests whenever SIL changes is a substantial and growing part of our job description.